docs/src/main/tut/guidelines.md

```diff
@@ -84,5 +84,32 @@ As an example, an implicit instance of `Monoid` for `List` defined in the packag
 
 This rule is relatively flexible. Use what you see appropriate. The goal is to maintain uniqueness and avoid conflicts.
 
+
+
+### <a id="implicit-priority" href="#implicit-priority"></a> Implicit instance priority
+
+When there are multiple instances provided implicitly, if the type class of them are in the same inheritance hierarchy, 
+the instances need to be separated out into different abstract class/traits so that they don't conflict with each other. The names of these abstract classes/traits should be numbered with a priority with 0 being the highest priority. The abstract classes/trait
+with higher priority inherits from the ones with lower priority. The most specific (whose type class is the lowest in the hierarchy) instance should be placed in the abstract class/ trait with the highest priority.  Here is an example. 
+
+```scala
+@typeclass 
+trait Functor[F[_]]
+
+@typeclass
+trait Monad[F[_]] extends Functor
+
+...
+object Kleisli extends KleisliInstance0 
+
+abstract class KleisliInstance0 extends KleisliInstance1 {
+  implicit def catsDataMonadForKleisli[F[_], A]: Monad[Kleisli[F, A, ?]] = ...
+}
+
+abstract class KleisliInstance1 {
+  implicit def catsDataFunctorForKleisli[F[_], A]: Functor[Kleisli[F, A, ?]] = ...
```

Comment:
It's really outside of the point of this example, but my mental compiler complained that you would need `F[_]:Monad` and `F[_]:Functor` constraints to be able to implement them :)